mmhuman3d/models/body_models/smplx.py

```diff
@@ -0,0 +1,247 @@
+                                           smplx_output.vertices)
+            joints = torch.cat([joints, extra_joints], dim=1)
+
+        joints, joint_mask = convert_kps(
+            joints,
+            src=self.keypoint_src,
+            dst=self.keypoint_dst,
+            approximate=self.keypoint_approximate)
+        if isinstance(joint_mask, np.ndarray):
+            joint_mask = torch.tensor(
+                joint_mask, dtype=torch.uint8, device=joints.device)
+
+        batch_size = joints.shape[0]
+        joint_mask = joint_mask.reshape(1, -1).expand(batch_size, -1)
+
+        output = dict(
+            global_orient=smplx_output.global_orient,
+            body_pose=smplx_output.body_pose,
+            joints=joints,
+            joint_mask=joint_mask,
+            keypoints=torch.cat([joints, joint_mask[:, :, None]], dim=-1),
+            betas=smplx_output.betas)
+
+        if return_verts:
+            output['vertices'] = smplx_output.vertices
+        if return_full_pose:
+            output['full_pose'] = smplx_output.full_pose
+
+        return output
+
+    @classmethod
+    def tensor2dict(cls,
+                    full_pose: torch.Tensor,
+                    betas: Optional[torch.Tensor] = None,
+                    transl: Optional[torch.Tensor] = None,
+                    expression: Optional[torch.Tensor] = None) -> dict:
+        """Convert full pose tensor to pose dict.
+
+        Args:
+            full_pose (torch.Tensor): shape should be (..., 165) or
+                (..., 55, 3). All zeros for T-pose.
+            betas (Optional[torch.Tensor], optional): shape should be
+                (..., 10). The batch num should be 1 or corresponds with
+                full_pose.
+                Defaults to None.
+            transl (Optional[torch.Tensor], optional): shape should be
+                (..., 3). The batch num should be 1 or corresponds with
+                full_pose.
+                Defaults to None.
+            expression (Optional[torch.Tensor], optional): shape should
+                be (..., 10). The batch num should be 1 or corresponds with
+                full_pose.
+                Defaults to None.
+
+        Returns:
+            dict: dict of smplx pose containing transl & betas.
+        """
+        NUM_BODY_JOINTS = cls.NUM_BODY_JOINTS
+        NUM_HAND_JOINTS = cls.NUM_HAND_JOINTS
+        NUM_FACE_JOINTS = cls.NUM_FACE_JOINTS
+        NUM_JOINTS = NUM_BODY_JOINTS + 2 * NUM_HAND_JOINTS + NUM_FACE_JOINTS
+        full_pose = full_pose.view(-1, (NUM_JOINTS + 1), 3)
+        global_orient = full_pose[:, :1]
+        body_pose = full_pose[:, 1:NUM_BODY_JOINTS + 1]
+        jaw_pose = full_pose[:, NUM_BODY_JOINTS + 1:NUM_BODY_JOINTS + 2]
+        leye_pose = full_pose[:, NUM_BODY_JOINTS + 2:NUM_BODY_JOINTS + 3]
+        reye_pose = full_pose[:, NUM_BODY_JOINTS + 3:NUM_BODY_JOINTS + 4]
+        left_hand_pose = full_pose[:, NUM_BODY_JOINTS + 4:NUM_BODY_JOINTS + 19]
```

Comment:
Replace `[:, NUM_BODY_JOINTS + 4:NUM_BODY_JOINTS + 19]` with '[:, NUM_BODY_JOINTS + 4:NUM_BODY_JOINTS + 4 + NUM_HAND_JOINTS]' if we want to avoid hard-code about hands.

Reply:
Refer to issue #62 .